modules/@apostrophecms/rich-text-widget/ui/apos/components/AposTiptapStyles.vue

```diff
@@ -57,26 +58,92 @@ export default {
       }
     }
   },
+  data() {
+    return {
+      multipleSelected: false
+    };
+  },
   computed: {
+    nodeOptions() {
+      return [ {
+        label: 'apostrophe:richTextNodeMultipleStyles',
+        attr: this.multipleSelected ? '' : 'hidden'
+      },
+      ...this.options.nodes ];
+    },
     active() {
-      for (let i = 0; (i < this.options.nodes.length); i++) {
-        const style = this.options.nodes[i];
-        if (this.editor.isActive(style.type, (style.options || {}))) {
-          return i;
-        } else if (this.editor.state.selection.$head.parent.type.name === 'defaultNode' && style.def) {
-          // Look deeper to see if custom defaultNode is active
-          return i;
+      const { selection } = this.editor.state;
+      const content = selection.content();
+      let activeEls = [];
+      const nodes = this.options.nodes.map(n => {
+        return {
+          type: n.type,
+          class: n.options.class || null,
+          level: n.options.level || null
+        };
+      });
+
+      if (content?.content?.content?.length) {
+        activeEls = content.content.content.map(n => {
+          return {
+            name: n.type.name,
+            class: n.attrs.class || null,
+            level: n.attrs.level || null
+          };
+        });
+      }
+
+      // Remove duplicates
+      activeEls = activeEls.filter((item, index, self) => {
+        // Find the index of the first occurrence of the current item
+        const firstIndex = self.findIndex(t =>
+          t.name === item.name &&
+          t.class === item.class &&
+          t.level === item.level
+        );
+        // If the index of the current item is the same as the first index, keep it
+        return index === firstIndex;
+      });
```

Comment:
recognize two separate nodes that are the same being selected are 'one' active type